sources/core/Stride.Core.CompilerServices/Analyzers/STRDIAG004PropertyWithNoGetter.cs

```diff
@@ -0,0 +1,60 @@
+using System.Collections.Immutable;
+using Stride.Core.CompilerServices.Common;
+
+namespace Stride.Core.CompilerServices.Analyzers;
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public class STRDIAG004PropertyWithNoGetter : DiagnosticAnalyzer
+{
+    public const string DiagnosticId = "STRDIAG004";
+    private const string Title = "Property with no Getter";
+    private const string MessageFormat = "The [DataMember] Attribute is applied to property '{0}' with an invalid getter accessibility level. Expected is a public/internal getter.";
```

Comment:
Can we adjust the message depending on issue?
```
The property '{0}' with [DataMember] does not have a getter which is required for serialization.
```
and if not accessible
```
The property '{0}' with [DataMember] does not have an accessible getter which is required for serialization. A public/internal getter is expected.
```

Reply:
[126a041](https://github.com/stride3d/stride/pull/1864/commits/126a041aa5e87ae73cd15990318b0319bebc094d)
now the diagnoser has 2 rules with your error messages